packages/astro/src/core/render/dev/index.ts

```diff
@@ -75,7 +83,7 @@ export async function render(renderers: SSRLoadedRenderer[], mod: ComponentInsta
 			children: '',
 		});
 		scripts.add({
-			props: { type: 'module', src: '/@id/astro/client/hmr.js' },
+			props: { type: 'module', src: new URL('../../../runtime/client/hmr.js', import.meta.url).pathname },
```

Comment:
this one is surprising! Any idea why this was required / what kind of error message you were seeing?

Reply:
It really is! I was seeing the following when Vite tried to resolve the module. Notice Astro returns a 404 as well:

```sh
14:23 PM [vite] Internal server error: Cannot set property 'isSelfAccepting' of undefined
  Plugin: vite:import-analysis
  File: /Users/benholmes/Sandbox/astro-integration-react/node_modules/.vite/astro_client_load_js.js?v=7a24946c
      at ModuleGraph.updateModuleInfo (/Users/benholmes/Sandbox/astro-integration-react/node_modules/vite/dist/node/chunks/dep-9c153816.js:53418:29)
      at TransformContext.transform (/Users/benholmes/Sandbox/astro-integration-react/node_modules/vite/dist/node/chunks/dep-9c153816.js:70072:57)
      at async Object.transform (/Users/benholmes/Sandbox/astro-integration-react/node_modules/vite/dist/node/chunks/dep-9c153816.js:38334:30)
      at async doTransform (/Users/benholmes/Sandbox/astro-integration-react/node_modules/vite/dist/node/chunks/dep-9c153816.js:53030:29)
7:14:23 PM [vite] Internal server error: Cannot set property 'isSelfAccepting' of undefined
  Plugin: vite:import-analysis
  File: /Users/benholmes/Sandbox/astro-integration-react/node_modules/.vite/astro_client_hmr_js.js?v=7a24946c
      at ModuleGraph.updateModuleInfo (/Users/benholmes/Sandbox/astro-integration-react/node_modules/vite/dist/node/chunks/dep-9c153816.js:53418:29)
      at TransformContext.transform (/Users/benholmes/Sandbox/astro-integration-react/node_modules/vite/dist/node/chunks/dep-9c153816.js:70072:57)
      at async Object.transform (/Users/benholmes/Sandbox/astro-integration-react/node_modules/vite/dist/node/chunks/dep-9c153816.js:38334:30)
      at async doTransform (/Users/benholmes/Sandbox/astro-integration-react/node_modules/vite/dist/node/chunks/dep-9c153816.js:53030:29)
```

Reply:
☝️ After tracing Vite's internal logs, there's a different `?v=XXXXX` attached to the end of the URL between calls. It seems to think the file changed even when it hasn't...